common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/SplunkLogStore.scala

```diff
@@ -137,10 +137,17 @@ class SplunkLogStore(
             ActivationLogs(
               r.results
                 .map(l =>
-                  //format same as org.apache.openwhisk.core.containerpool.logging.LogLine.toFormattedString
-                  f"${l.fields(splunkConfig.logTimestampField).convertTo[String]}%-30s ${l
-                    .fields(splunkConfig.logStreamField)
-                    .convertTo[String]}: ${l.fields(splunkConfig.logMessageField).convertTo[String].trim}"))
+                  try {
```

Comment:
Instead of using simple try...catch you can use [Scala Try][1]. This ensures that you do not eat up fatal exception and also handle the failure case in bit more functional way!

```suggestion
 r.results
                  .map(js => Try(toLogLine(js)))
                  .map {
                    case Success(s) => s
                    case Failure(t) =>
                      logging.debug(
                        this,
                        s"The log message might have been too large " +
                          s"for '${splunkConfig.index}' Splunk index and can't be retrieved, ${t.getMessage}")
                      s"The log message can't be retrieved, ${t.getMessage}"
                  })
```

[1]: https://danielwestheide.com/blog/2012/12/26/the-neophytes-guide-to-scala-part-6-error-handling-with-try.html

Reply:
Thx for the tip @chetanmeh!